openpathsampling/tests/analysis/utils.py

```diff
@@ -0,0 +1,500 @@
+        error_msg="mover matching signature{sig}".format(sig=signature)
+    )
+    return mover
+
+def _run_patched(mover, patches, inputs):
+    """Run the move with the given patches for the given inputs"""
+    for patch in patches:
+        patch.start()
+
+    change = mover.move(inputs)
+
+    for patch in patches:
+        patch.stop()
+    return change
+
+
+class MockRandomChoiceMover(object):
+    """Mock random choice mover to give the desired subchange.
+
+    This differs from the regular mover mocks in that it is only created
+    when we have already determined the subchange that will be generated.
+
+    Parameters
+    ----------
+    random_mover: class:`.RandomChoiceMover`
+        the mover to mock the behavior or
+    change : :class:`.MoveChange`
+        the change that the submover would return
+    """
+    def __init__(self, random_mover, change):
+        self.random_mover = random_mover
+        self.change = change
+
+    def patches(self):
+        mover = self.change.mover
+        try:
+            idx = self.random_mover.submovers.index(mover)
+        except ValueError:
+            raise AnalysisTestSetupError(
+                "mover %s not found as submover of mover %s" %
+                (self.random_mover, mover)
+            )
+        rng_mock = Mock(choice=Mock(return_value=idx))
+        patches = [
+            patch.object(self.random_mover, '_rng', rng_mock),
+            patch.object(mover, 'move', Mock(return_value=self.change))
+        ]
+        return patches
+
+    def __call__(self, inputs):
+        patches = self.patches()
+        return _run_patched(self.random_mover, patches, inputs)
+
+
+class MockMove(object):
+    """Abstract object for mocking a single step of a path simulation.
+    """
+    def __init__(self, scheme, ensembles=None, group_name=None):
+        self.scheme = scheme
+        self.ensembles = ensembles
+        self.group_name = group_name
+
+    def wrap_org_by_group(self, change, inputs):
+        """Wraps as if using an OrganizeByMoveGroup strategy
+        """
+        # extract the root_mover (selects type of move) and the
+        # group_selector (selects a specific move within the move type)
+        root_mover = self.scheme.root_mover
+        group_selector = _get_only(
+            iterable=root_mover.submovers,
+            condition=lambda g: change.mover in g.submovers,
+            error_msg="group containing the mover {mover}".format(
+                mover=change.mover
+            )
+        )
+
+        # make a move change for the inner step (selecting which mover
+        # within the move type)
+        inner_mock = MockRandomChoiceMover(group_selector, change)
+        inner_change = inner_mock(inputs)
+
+        # make a move change for the outer step (selecting which move type
+        # to do from the root_mover)
+        outer_mock = MockRandomChoiceMover(root_mover, inner_change)
+        outer_change=  outer_mock(inputs)
```

Comment:
whitespace complaint
```suggestion
        outer_change = outer_mock(inputs)
```